dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/ContentletFactoryTest.java

```diff
@@ -63,33 +70,46 @@ public void findAllCurrent () throws DotDataException, DotSecurityException {
     @Test
     public void findAllCurrentOffsetLimit () throws DotDataException, DotSecurityException {
 
-        //Getting all contentlets live/working contentlets
-        List<Contentlet> contentlets = contentletFactory.findAllCurrent( 0, 5 );
-
-        //Validations
-        assertTrue( contentlets != null && !contentlets.isEmpty() );
-        assertEquals( contentlets.size(), 5 );
-
-        //Validate the integrity of the array
-        Contentlet foundContentlet = null;
-        for ( Contentlet contentlet : contentlets ) {
-
-            //TODO: We need to verify for null because the findAllCurrent CAN return null objects, this could happen because the index can return inodes that are not into the db....
-            if ( contentlet != null ) {
-                foundContentlet = contentlet;
-                break;
+        final ContentType type = new ContentTypeDataGen().nextPersisted();
+        final ContentletDataGen contentletDataGen = new ContentletDataGen(type.id());
+        final List<Contentlet> newContentlets = new ArrayList<>();
+
+        try {
+
+            // Let's create 10 contentlets, 5 live 5 working
+            IntStream.range(0, 10).forEach(
+                    (i) -> {
+                        final Contentlet newContent = contentletDataGen.nextPersisted();
+                        if (i % 2 == 0) {
+                            Sneaky.sneaked(() -> APILocator.getContentletAPI()
+                                    .publish(newContent, user, false));
+                        }
+                        newContentlets.add(newContent);
+                    }
+            );
+
+            //Getting all contentlets live/working contentlets
+            List<Contentlet> contentlets = contentletFactory.findAllCurrent(0, 10);
+
+            // filter out null records
+            contentlets = contentlets.stream().filter(Objects::nonNull).collect(Collectors.toList());
+            //Validations
+            assertTrue(contentlets != null && !contentlets.isEmpty());
+            assertTrue(contentlets.size() >= 10);
+
+            //Search for one of the objects we found
+            String inode = contentlets.get(0).getInode();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'inode' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4884522)